app/src/main/java/com/kickstarter/viewmodels/ThreadViewModel.kt

```diff
@@ -69,13 +82,53 @@ interface ThreadViewModel {
             comment
                 .compose(bindToLifecycle())
                 .subscribe(this.rootComment)
+
+            val project = intent()
+                .map { it.getParcelableExtra(IntentKey.PROJECT) as Project? }
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+
+            val loggedInUser = this.currentUser.loggedInUser()
+                .filter { u -> u != null }
+                .map { requireNotNull(it) }
+
+            loggedInUser
+                .compose(bindToLifecycle())
+                .subscribe {
+                    currentUserAvatar.onNext(it.avatar().small())
+                }
+
+            loggedInUser
+                .compose(bindToLifecycle())
+                .subscribe {
+                    showReplayComposer.onNext(true)
+                }
+
+            project
+                .compose(Transformers.combineLatestPair(currentUser.observable()))
+                .compose(bindToLifecycle())
+                .subscribe {
+                    val composerStatus = getCommentComposerStatus(Pair(it.first, it.second))
+                    showReplayComposer.onNext(composerStatus != CommentComposerStatus.GONE)
+                    replayComposerStatus.onNext(composerStatus)
+                }
         }
 
+        private fun getCommentComposerStatus(projectAndUser: Pair<Project, User?>) =
```

Comment:
I wonder if we can abstract this logic into something separate since this is being reused in a couple places? Just a thought.